openmmtools/multistate/multistatereporter.py

```diff
@@ -362,13 +362,6 @@ def _open_dataset_robustly(self, *args, n_attempts=5, sleep_time=2,
         for attempt in range(n_attempts-1):
             try:
                 return netcdf.Dataset(*args, **kwargs)
-            except IOError as e:
-                # If the file does not exist, it doesn't make sense to try again.
-                if catch_io_error:
-                    if io_error_warning is not None:
-                        logger.warning(io_error_warning)
-                    return None
-                raise e
             except:
                 logger.debug('Attempt {}/{} to open {} failed. Retrying '
                              'in {} seconds'.format(attempt+1, n_attempts, sleep_time))
```

Comment:
There are four `{}` items in this string but only three arguments passed, which is now causing test failures:
```
======================================================================
ERROR: Test that the analysis file can open without the checkpoint file
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/miniconda/envs/test/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/runner/work/openmmtools/openmmtools/openmmtools/tests/test_sampling.py", line 1340, in test_analysis_opens_without_checkpoint
    self.REPORTER(storage_path, checkpoint_storage=cp_file_mod, open_mode='r')
  File "/Users/runner/work/openmmtools/openmmtools/openmmtools/multistate/multistatereporter.py", line 136, in __init__
    self.open(open_mode)
  File "/Users/runner/work/openmmtools/openmmtools/openmmtools/multistate/multistatereporter.py", line 302, in open
    version=netcdf_format)
  File "/Users/runner/work/openmmtools/openmmtools/openmmtools/multistate/multistatereporter.py", line 367, in _open_dataset_robustly
    'in {} seconds'.format(attempt+1, n_attempts, sleep_time))
IndexError: tuple index out of range
```

Reply:
Your right, but what is weird is the code I removed didn't change that, so this exception would never have worked. I'm going to fix the code + double check what the tests are doing now as well as remove those defunct kwargs.